test/DualTest.jl

```diff
@@ -499,10 +521,14 @@ for N in (0,3), M in (0,4), V in (Int, Float32)
                     else
                         @test dx isa Complex{<:Dual{TestTag()}}
                         @test dy isa Complex{<:Dual{TestTag()}}
-                        @test real(value(dx)) == real(actualval)
-                        @test real(value(dy)) == real(actualval)
-                        @test imag(value(dx)) == imag(actualval)
-                        @test imag(value(dy)) == imag(actualval)
```

Comment:
These tests use method `value(x) = x`.

But they look like they are written to use a method `value(::Complex{<:Dual})`, which doesn't exist.

Reply:
Good catch, but it actually did the correct thing nevertheless - only changing the definition of `==` in this PR brakes it :stuck_out_tongue: `value(x::Complex{<:Dual}) = x`, so it's a no-op, and then `real` and `imag` just extract the different duals. Currently, then `somedual == somefloat` just checks `value(some_dual) == somefloat` (as intended in this test) but this PR change it to `value(some_dual) == somefloat && iszero(partials(some_dual))` which, of course, is not true in general and causes test errors.

Reply:
So I discovered. The question is whether there ought to be `value(::Complex{<:Dual})`, probably yes, but probably not this PR.